### PR TITLE
Ready: Basic docs for the most commonly used apis.

### PR DESCRIPTION
**Pull request by alexmoore**

### CODE REVIEW

File: src/RiakClient/Extensions/JsonExtensions.cs [hunk 1]
```diff
@@ -21,18 +21,42 @@ namespace RiakClient.Extensions
 {
     using Newtonsoft.Json;
 
+    /// <summary>
+    /// Helper extension methods to use common Newtonsoft.Json methods with less code. 
+    /// </summary>
     public static class JsonExtensions
```
Comment (lukebakken) [inline src/RiakClient/Extensions/JsonExtensions.cs]: How about `internal`? Would this then need documentation?

File: src/RiakClient/Extensions/StringExtensions.cs [hunk 2]
```diff
@@ -24,7 +24,10 @@ namespace RiakClient.Extensions
     using System.Web;
     using Util;
 
-    internal static class StringExtensions
+    /// <summary>
+    /// Extension methods to assist with conversion between Riak byte[] and strings.
+    /// </summary>
+    public static class StringExtensions
```
Comment (lukebakken) [inline src/RiakClient/Extensions/StringExtensions.cs]: This should probably be `internal` too. No reason for library users to be using our extension methods.

File: src/RiakClient/IRiakAsyncClient.cs [hunk 3]
```diff
@@ -28,74 +28,718 @@ namespace RiakClient
...
+        /// <description>
+        /// Ping can be used to ensure that there is an operational Riak node
+        /// present at the other end of the client. It's important to note that this will ping
+        /// any Riak node in the cluster and a specific node cannot be specified by the user.
+        /// Do not use this method to determine individual node health.
+        /// </description>
+        /// <returns>
+        /// A <see cref="Task{TResult}"/> that will contain a <see cref="RiakResult"/>. 
+        /// Check <see cref="RiakResult.IsSuccess"/> to see if the operation was a success.
+        /// </returns>
```
Comment (lukebakken) [inline src/RiakClient/IRiakAsyncClient.cs]: Epic docs very nice.

File: src/RiakClient/IRiakAsyncClient.cs [hunk 4]
```diff
@@ -28,74 +28,718 @@ namespace RiakClient
...
+
+        /// <summary>
+        /// Used to create a batched set of actions to be sent to a Riak cluster.
+        /// </summary>
+        /// <typeparam name="T">The <paramref name="batchFun"/>'s return type.</typeparam>
+        /// <param name="batchFunction">A func that wraps all the operations to batch together.</param>
+        /// <returns>A <see cref="Task{TResult}"/> that will contain the return value of <paramref name="batchFun"/>.</returns>
+        Task<T> Batch<T>(Func<IRiakBatchClient, T> batchFunction);
+
+        // TODO: We're missing everything below:
```
Comment (lukebakken) [inline src/RiakClient/IRiakAsyncClient.cs]: Just delete it?

File: src/RiakClient/Models/RiakObject.cs [hunk 5]
```diff
@@ -343,51 +457,140 @@ public BinIndex BinIndex(string name)
...
+        /// <summary>
+        /// Create a link from this object to another using a tag.
+        /// </summary>
+        /// <param name="riakObject">The other object.</param>
+        /// <param name="tag">The tag to use.</param>
+        /// <exception cref="RiakUnsupportedException">
+        /// Thrown if RiakObjectId has a bucket type. 
+        /// Combining linkwalking and bucket types is not supported.
+        /// </exception>
+        [Obsolete("Linkwalking has been depreciated as of Riak 2.0. This method will be removed in the next major version.")]
```
Comment (lukebakken) [inline src/RiakClient/Models/RiakObject.cs]: depreciated -> deprecated

Comment (alexmoore) [inline src/RiakClient/Models/RiakObject.cs]: Why don't you apprecate my spelling? :D

File: src/RiakClient/Extensions/StringExtensions.cs (hunk 2 above)
Comment (alexmoore) [inline src/RiakClient/Extensions/StringExtensions.cs]: I think we have users that use the `ToRiakString/FromRiakString` methods, as it's the easy way to serialize.

File: src/RiakClient/Extensions/JsonExtensions.cs (hunk 1 above)
Comment (alexmoore) [inline src/RiakClient/Extensions/JsonExtensions.cs]: With internal, no docs needed.  Not sure if we have any customers using it though.

File: src/RiakClient/Extensions/StringExtensions.cs (hunk 2 above)
Comment (lukebakken) [inline src/RiakClient/Extensions/StringExtensions.cs]: OK cool

**Comment by lukebakken** [comment]
@alexmoore I'm assuming you get a bunch of these warnings:

```
C:\Program Files (x86)\EWSoftware\Sandcastle Help File Builder\SandcastleHelpFileBuilder.targets(40,3): warning : BuildAssembler : warning : ShowMissingComponent: [M:RiakClient.Timeo ut.ToString] Missing <returns> documentation [C:\Users\lbakken\Projects\basho\riak-dotnet-client\src\RiakClient\docs\Working\BuildReferenceTopics.proj] [C:\Users\lbakken\Projects\basho\riak-d otnet-client\src\RiakClient.shfbproj]
```

**Comment by lukebakken** [comment]
FYI:
https://github.com/basho-labs/riak-dotnet-client-api

**Comment by lukebakken** [comment]
:shipit:

**Pull request merged**